akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/JsonStreamingExamplesSpec.scala

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.scaladsl.server.directives
+
+import akka.NotUsed
+import akka.http.scaladsl.marshalling._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.Accept
+import akka.http.scaladsl.server.{ EntityStreamingSupport, JsonEntityStreamingSupport, UnacceptedResponseContentTypeRejection, UnsupportedRequestContentTypeRejection }
+import akka.stream.scaladsl.{ Flow, Source }
+import akka.util.ByteString
+import docs.http.scaladsl.server.RoutingSpec
+import spray.json.JsValue
+
+import scala.concurrent.Future
+
+class JsonStreamingExamplesSpec extends RoutingSpec {
+
+  //#models
+  case class Tweet(uid: Int, txt: String)
+  case class Measurement(id: String, value: Int)
+  //#
+
+  def getTweets() =
+    Source(List(
+      Tweet(1, "#Akka rocks!"),
+      Tweet(2, "Streaming is so hot right now!"),
+      Tweet(3, "You cannot enter the same river twice.")))
+
+  //#formats
+  object MyJsonProtocol
+    extends akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+    with spray.json.DefaultJsonProtocol {
+
+    implicit val tweetFormat = jsonFormat2(Tweet.apply)
+    implicit val measurementFormat = jsonFormat2(Measurement.apply)
+  }
+  //#
+
+  "spray-json-response-streaming" in {
+    // [1] import "my protocol", for marshalling Tweet objects:
+    import MyJsonProtocol._
+
+    // [2] pick a Source rendering support trait:
+    implicit val jsonStreamingSupport: JsonEntityStreamingSupport = EntityStreamingSupport.json()
```

Comment:
We can ask about feedback on the default since this is an RC release today.
I've seen most streaming APIs do the newline separated somehow (while technically it's wrong, yeah).

https://www.flowdock.com/api/streaming
https://dev.twitter.com/streaming/reference/get/statuses/sample

I guess you may be right though that array is the more typical case. Not everyone is twitter.